tests/post/steps/test_monitoring.py

```diff
@@ -119,10 +134,152 @@ def _check_object_exists():
+        "apiVersion": "v1",
+        "metadata": {
+            "name": "zk-dash",
+            "labels": {
+                "app.kubernetes.io/managed-by": "salt",
+                "app.kubernetes.io/part-of": "zenko",
+                "grafana_dashboard": "1",
+                "heritage": "salt",
+            },
+        },
+        "data": {
+            f"{cm_uid}-zk-dash.json": json.dumps(
+                {
+                    "annotations": {"list": []},
+                    "description": "MongoDB Dashboard.",
+                    "editable": True,
+                    "graphTooltip": 1,
+                    "links": [],
+                    "panels": [],
+                    "refresh": "5s",
+                    "schemaVersion": 30,
+                    "style": "dark",
+                    "tags": [],
+                    "templating": {"list": []},
+                    "timezone": "browser",
+                    "title": "Zenko Dashboard",
+                    "version": 1,
+                    "uid": cm_uid,
+                }
+            )
+        },
+    }
+    context["grafana-cm-manifest"] = cm_manifest
+
+
+# }}}
+# When {{{
+
+
+@when(
+    parsers.parse(
+        "We put the dashboard as a ConfigMap in namespace '{namespace}' with folder annotation value '{cm_folder}'"
+    )
+)
+def push_dashboard_cm(k8s_client, context, cm_folder, namespace):
+    client = kube_utils.Client(k8s_client, "ConfigMap", namespace=namespace)
+    configmaps = client.list()
+
+    cm_manifest = context["grafana-cm-manifest"]
+    cm_name = cm_manifest.get("metadata", {}).get("name")
+
+    cm_manifest["metadata"]["annotations"] = {
+        "metalk8s.scality.com/grafana-folder-name": cm_folder,
+    }
+    cm_manifest["metadata"]["namespace"] = namespace
+
+    try:
+        next(
+            filter(lambda cm: cm.get("metadata", {}).get("name") == cm_name, configmaps)
+        )
+        try:
+            client.delete(name=cm_name)
+        except ApiException as err:
+            raise AssertionError(f"Unable to delete existing ConfigMap: {err}")
+    except StopIteration:
+        pass
+    finally:
+        try:
+            client._create(cm_manifest)
+        except ApiException as err:
+            raise AssertionError(f"Unable to push ConfigMap: {err}")
+
+
 # }}}
 # Then {{{
 
 
+@then(
+    parsers.parse("on deletion of the dashboard's ConfigMap from namespace '{ns_name}'")
+)
+def delete_config_map(k8s_client, context, ns_name):
+    client = kube_utils.Client(k8s_client, "ConfigMap", namespace=ns_name)
+    cm_manifest = context["grafana-cm-manifest"]
+    cm_name = cm_manifest.get("metadata", {}).get("name")
+    try:
+        client.delete(cm_name)
+    except ApiException as err:
+        raise AssertionError(f"failed to delete dashboard '{cm_name}': {err}")
+
+
+@then(parsers.parse("the dashboard is no longer in folder '{folder_name}' in Grafana"))
+def check_no_dash_in_folder(context, gf_folder_name, grafana_api):
```

Comment:
Should be:

```suggestion
@then(parsers.parse("the dashboard is no longer in folder '{gf_folder_name}' in Grafana"))
def check_no_dash_in_folder(context, gf_folder_name, grafana_api):
```

I imagine?

Reply:
Oops, fixed in `c60bea50e`